src/Text/Pandoc/Writers/Txt2Tags.hs

```diff
@@ -0,0 +1,512 @@
+
+inlineToTxt2Tags opts (Underline lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "__" <> contents <> "__"
+
+inlineToTxt2Tags opts (Strong lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "**" <> contents <> "**"
+
+inlineToTxt2Tags opts (Strikeout lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "--" <> contents <> "--"
+
+inlineToTxt2Tags opts (Superscript lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "<sup>" <> contents <> "</sup>"
+
+inlineToTxt2Tags opts (Subscript lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "<sub>" <> contents <> "</sub>"
+
+inlineToTxt2Tags opts (SmallCaps lst) = inlineListToTxt2Tags opts lst
+
+inlineToTxt2Tags opts (Quoted SingleQuote lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "\8216" <> contents <> "\8217"
+
+inlineToTxt2Tags opts (Quoted DoubleQuote lst) = do
+  contents <- inlineListToTxt2Tags opts lst
+  return $ "\8220" <> contents <> "\8221"
+
+inlineToTxt2Tags opts (Cite _  lst) = inlineListToTxt2Tags opts lst
+
+inlineToTxt2Tags _ (Code _ str) =
+  -- In Txt2Tags, text surrounded by '' is really just a font statement, i.e. <tt>,
+  -- and so other formatting can be present inside.
+  -- However, in pandoc, and markdown, inlined code doesn't contain formatting.
+  -- So I have opted for using %% to disable all formatting inside inline code blocks.
+  -- This gives the best results when converting from other formats to Txt2Tags, even if
+  -- the resultand code is a little ugly, for short strings that don't contain formatting
+  -- characters.
+  -- It does mean that if pandoc could ever read Txt2Tags, and so round-trip the format,
+  -- any formatting inside inlined code blocks would be lost, or presented incorrectly.
+  return $ "''%%" <> str <> "%%''"
+
+inlineToTxt2Tags _ (Str str) = return $ escapeString str
+
+inlineToTxt2Tags _ (Math mathType str) = return $ delim <> str <> delim
+                                 -- note:  str should NOT be escaped
+  where delim = case mathType of
+                     DisplayMath -> "$$"
+                     InlineMath  -> "$"
+
+inlineToTxt2Tags _ il@(RawInline f str)
+  | f == Format "Txt2Tags" = return str
+  | f == Format "html"     = return $ "<html>" <> str <> "</html>"
+  | otherwise              = "" <$ report (InlineNotRendered il)
+
+inlineToTxt2Tags _ LineBreak = do
+  backSlash <- asks stBackSlashLB
+  return $ if backSlash
+           then "\n"
+           else "\\\\\n"
+
+inlineToTxt2Tags opts SoftBreak =
+  case writerWrapText opts of
+       WrapNone     -> return " "
+       WrapAuto     -> return " "
+       WrapPreserve -> return "\n"
+
+inlineToTxt2Tags _ Space = return " "
+
+inlineToTxt2Tags opts (Link _ txt (src, _)) = do
+  label <- inlineListToTxt2Tags opts txt
+  case txt of
+     [Str s] | "mailto:" `T.isPrefixOf` src -> return $ "<" <> s <> ">"
+             | escapeURI s == src -> return src
+     _  -> if isURI src
```

Comment:
Is this `if` statement necessary? I'm not aware of a string with a leading `/` that would be recognized as valid URI.

Reply:
it's just copied from Dokuwiki... most txt2tags syntax is similar to dokuwiki, so I suppose it's legit

![image](https://user-images.githubusercontent.com/477243/104966145-7b3be000-59e0-11eb-9df1-9af0805f1a1c.png)

probably the code should be refined in the future